builtin/aws/ecs/task.go

```diff
@@ -362,4 +499,60 @@ func roleArn(name string, sess *session.Session) (string, error) {
 	return *roleOut.Role.Arn, nil
 }
 
+// taskStatus gets the status of an ECS task and provides it to the caller via a channel
+// the caller is responsible for closing the channel
+func taskStatus(ctx context.Context, d time.Duration, taskStatusCh *chan string, log hclog.Logger, sess *session.Session, cluster string, taskId string) {
+	ecsSvc := ecs.New(sess)
+
+	taskContext, cancel := context.WithTimeout(ctx, d)
+	defer cancel()
+	for {
+		select {
+		case <-taskContext.Done():
+			log.Error("timeout waiting for task to become ready", terminal.WithErrorStyle())
+			return
+		default:
```

Comment:
Since the default is essentially "execute the loop", maybe we only have 1 select case for when we timeout, then outside of the select, do this logic?